listenbrainz/webserver/views/user_setting_api.py

```diff
@@ -0,0 +1,36 @@
+import ujson
+
+from brainzutils.ratelimit import ratelimit
+from flask import Blueprint, jsonify, request
+
+import listenbrainz.db.user as db_user
+import listenbrainz.db.user_setting as db_usersetting
+from listenbrainz.db.exceptions import DatabaseException
+from listenbrainz.webserver.decorators import crossdomain
+from listenbrainz.webserver.errors import APIInternalServerError,APIBadRequest
+
+
+user_setting_api_bp = Blueprint('user_setting_api_v1', __name__)
+
+
+@user_setting_api_bp.route('/<user_name>/timezone', methods=["POST", "OPTIONS"])
+@crossdomain
+@ratelimit()
+def reset_timezone(user_name):
```

Comment:
For normal views, `@login_required` performs the authentication but for API it has to be done manually by reading the Authorization header of the request. We have a helper method for this:

https://github.com/metabrainz/listenbrainz-server/blob/f20cedc430b29600674ce2d83badb42dc3db9f53/listenbrainz/webserver/views/api_tools.py#L459

Example usage is simple, just call it:
https://github.com/metabrainz/listenbrainz-server/blob/f20cedc430b29600674ce2d83badb42dc3db9f53/listenbrainz/webserver/views/pinned_recording_api.py#L48

The helper will also return the user who possesses the token in Authorization header.

But now if you have username in the url as well, then you have to verify that the token matches that user. So, I would suggest to get rid of the username from the url. `/1/user-setting/timezone` is probably enough.

Reply:
Thanks, Lucifer! I made the change in my new commit. Also, the user's timezone will be saved and updated on the page after clicking the `save timezone` button. Please take a look at it.

Reply:
Thanks